lib/package-managers/npm.js

```diff
@@ -21,19 +22,27 @@ function parseJson(result, data) {
 
 /**
  * @param packageName   Name of the package
- * @param field         Field such as "versions" or "dist-tags.latest" accepted by npm view (https://docs.npmjs.com/api/view)
- * @param {Object} options
+ * @param field         Field such as "versions" or "dist-tags.latest" are parsed from the package-json result (https://github.com/sindresorhus/package-json)
  * @Returns             Promised result
  */
-function view(packageName, field, options) {
-    const args = ['view', packageName, field];
-    return spawnNpm(args, options)
-        .then(result => {
-            return parseJson(result, {
-                command: `npm ${args.join(' ')}`,
-                packageName
-            });
-        });
+function view(packageName, field) {
+    let options = {allVersions: true};
+    if (field === 'time') {
+        options.fullMetadata = true;
+    }
+    return packageJson(packageName, options).then(result => {
+        if (field.startsWith('dist-tags.')) {
+            const split = field.split('.');
+            return result[split[0]][split[1]];
+        } else if (field === 'versions') {
+            return Object.keys(result[field]);
+        } else {
+            return result[field];
+        }
+    }).catch(error => {
```

Comment:
Unless I am missing something, we should remove the `catch` and allow errors to bubble up here. If an error needs to be ignored, we should match the text of the error message. That said, it should probably still throw and allow it to be caught higher up if necessary.

Reply:
Thanks for the fast review! I've added the catch because there are [two tests](https://github.com/tjunnone/npm-check-updates/blob/49e258a6c96b497aec6fb0796a54f6b69f81ff0d/test/test-versionmanager.js#L420-L428) which specify that unknown packages should be ignored, thus returning `false`.

Though I realized that using `console.error` would mess with `--jsonAll` and `--silent` - I've tweaked it to check for `package-json`'s exported `PackageNotFoundError` and throw in any other case.

Reply:
@stoically FYI I decided to leverage the existing 404 handling: https://github.com/tjunnone/npm-check-updates/commit/529bebad3e4e1e2f26c328d15dc5597f41745995

Thanks so much for making this happen! A huge improvement.